packages/tao/src/shared/workspace.ts

```diff
@@ -762,6 +762,7 @@ function buildProjectConfigurationFromPackageJson(
     root: directory,
     sourceRoot: directory,
     name,
+    implicitDependencies: packageJson?.nx?.implicitDependencies,
```

Comment:
Rather than pull out the implicitDependencies here, we should do something like:

```typescript
const {targets, ...rest} = packageJson.nx ?? {};
return {
  root: directory,
  ...rest,
  sourceRoot: rest.sourceRoot ?? directory
}
```

Reply:
We have to pull out the targets from the object, otherwise, target inference may get a little messed up. We should check that though, if everything works fine without the `targets` object removed we can keep it in.

Reply:
Thanks for the feedback. If targets are pulled out, how will they be taken into consideration given they won't be present in the Project Configuration?